src/utils/gauss_quad.hpp

```diff
@@ -0,0 +1,619 @@
+//  Copyright John Maddock 2015.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef GAUSS_QUAD_HPP
+#define GAUSS_QUAD_HPP
+
+#include <array>
+#include <cmath>
+#include <limits>
+
+#include "../kokkos_abstraction.hpp"
+
+namespace parthenon::math::quadrature {
+namespace detail {
+
+template <class T>
+struct gauss_constant_category {
+  static const unsigned value =
+      (std::numeric_limits<T>::is_specialized == 0) ? 999
```

Comment:
we maybe want to give `999` a name here to describe intent (i.e., that it's an invalid choice for the category). Similarly maybe we want to give 0, 1, 2 names for float, double, long double

Reply:
agreed, will change this.